### PR TITLE
feat(oas): implement mocker

### PR DESCRIPTION
**Pull request by sixaphone**
* Added method to mock data passed on passed data type in converter and call method
* Export method for testing
* Implement tests for mocker

Solves #4

### CODE REVIEW

File: src/converter.js [hunk 1]
```diff
@@ -375,6 +375,34 @@ var resolveRef = function(oai, ref) {
...
+      return true;
+    case 'ARRAY':
+      return ['lorem', 2, true, {key: 'value'}];
+    case 'OBJECT':
+      return ({
+        key: 'value'
+      });
+  }
+  return '';
+}
```
Comment (ArtemDerevnjuk) [inline src/converter.js]: You doesn't cover following types:
* `integer`
* `null`

I couldn't ignore type restrictions:
* **string**: `pattern`, `format`, `enum`, `minLength` and `maxLength`
* **number** and **integer**: `minimum`, `maximum`, `exclusiveMaximum`, `exclusiveMinimum`, `multipleOf`, `enum`
* **array**: `items`, `minItems`, `maxItems`, `uniqueItems`, `additionalItems` (`items` types should be handled recursively)
* **object**: `required`, `properties`, `propertyNames`, `minProperties`, `maxProperties`, `patternProperties`

File: src/converter.js [hunk 2]
```diff
@@ -158,7 +158,7 @@ var getQueryStrings = function(swagger, path, method, values) {
           name: param.name,
           value: typeof values[param.name] === "undefined"
             ? (typeof param.default === "undefined"
-              ? ("SOME_" + (param.type || param.schema.type).toUpperCase() + "_VALUE")
+              ? getMockData((param.type || param.schema.type).toUpperCase())
```
Comment (ArtemDerevnjuk) [inline src/converter.js]: Don't need to call `.toUpperCase()`

File: src/converter.js (hunk 1 above)
Comment (ArtemDerevnjuk) [reply src/converter.js]: Use `example`, if it's defined

File: .github/workflows/config.yml [hunk 3]
```diff
@@ -18,6 +18,7 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
       - run: npm ci
+      - run: npm run build
```
Comment (ArtemDerevnjuk) [inline .github/workflows/config.yml]: See `package.json`. We didn't define `build` script

File: release.config.js [hunk 4]
```diff
@@ -16,17 +19,18 @@ module.exports = {
         }
       }
     ],
+    '@semantic-release/npm',
     [
-      '@semantic-release/npm',
+      '@semantic-release/github',
       {
-        npmPublish: false
+        assets: ['dist/**', 'src/**', 'package.json', 'package-lock.json', 'README.md']
```
Comment (ArtemDerevnjuk) [inline release.config.js]: We don't generate artifacts for this package

File: release.config.js [hunk 5]
```diff
@@ -16,17 +19,18 @@ module.exports = {
...
     ],
     [
-      '@semantic-release/github',
-      {
-        assets: ['dist/**']
-      }
+      '@semantic-release/git', {
+      assets: ['src/**', 'package.json', 'package-lock.json'],
+      message: 'chore(release): ${nextRelease.version} [skip ci]'
+    }
```
Comment (ArtemDerevnjuk) [inline release.config.js]: Could you format json?

File: src/converter.js [hunk 6]
```diff
@@ -148,19 +149,16 @@ var getQueryStrings = function(swagger, path, method, values) {
...
       if (typeof param.in !== 'undefined' && param.in.toLowerCase() === 'query') {
+        const sample = OpenAPISampler.sample(param.schema || param, {});
         queryStrings.push({
           name: param.name,
-          value: typeof values[param.name] === 'undefined'
-            ? (typeof param.default === 'undefined'
-              ? ('SOME_' + (param.type || param.schema.type).toUpperCase() + '_VALUE')
-              : param.default + '')
-            : (values[param.name] + '') /* adding a empty string to convert to string */
+          value: JSON.stringify(sample)
```
Comment (ArtemDerevnjuk) [inline src/converter.js]: You shouldn't ignore `values` arguments and just use `sample` as fallback values.

Comment (ArtemDerevnjuk) [inline src/converter.js]: You should call `.stringify()`  only when `sample` is object. Any case you should use `encodeURIComponent` to escape special characters.

File: src/converter.js [hunk 7]
```diff
@@ -220,9 +218,10 @@ var getHeadersArray = function(swagger, path, method) {
     for (var k in pathObj.parameters) {
       var param = pathObj.parameters[k]
       if (typeof param.in !== 'undefined' && param.in.toLowerCase() === 'header') {
+        const sample = OpenAPISampler.sample(param.schema || param, {});
         headers.push({
           name: param.name,
-          value: 'SOME_' + (param.type || param.schema.type).toUpperCase() + '_VALUE'
+          value: JSON.stringify(sample)
```
Comment (ArtemDerevnjuk) [inline src/converter.js]: See my remarks for `queryStrings`.

File: src/converter.js [hunk 8]
```diff
@@ -148,19 +149,16 @@ var getQueryStrings = function(swagger, path, method, values) {
...
   if (typeof swagger.paths[path][method].parameters !== 'undefined') {
     for (var i in swagger.paths[path][method].parameters) {
-      var param = swagger.paths[path][method].parameters[i]
+      var param = swagger.paths[path][method].parameters[i];
       if (typeof param['$ref'] === 'string' &&
         !/^http/.test(param['$ref'])) {
         param = resolveRef(swagger, param['$ref'])
       }
       if (typeof param.in !== 'undefined' && param.in.toLowerCase() === 'query') {
+        const sample = OpenAPISampler.sample(param.schema || param, {});
```
Comment (ArtemDerevnjuk) [inline src/converter.js]: You should pass `definitions` as third argument into `.sample()`. Otherwise, sample can't find schemes by json pointer.

**Comment by ArtemDerevnjuk** [comment]
@sixaphone hmm... we should install `@semantic-release/git`. Seems it's missed in `package.json`

File: src/converter.js [hunk 9]
```diff
@@ -218,10 +221,10 @@ var getHeadersArray = function(swagger, path, method) {
     for (var k in pathObj.parameters) {
       var param = pathObj.parameters[k]
       if (typeof param.in !== 'undefined' && param.in.toLowerCase() === 'header') {
-        const sample = OpenAPISampler.sample(param.schema || param, {});
+        const sample = OpenAPISampler.sample(param.schema || param, {}, swagger);
         headers.push({
           name: param.name,
-          value: JSON.stringify(sample)
+          value: encodeURIComponent((typeof sample === 'object') ? JSON.stringify(sample) : sample)
```
Comment (ArtemDerevnjuk) [inline src/converter.js]: Don't need to use `encodeURIComponent`. Http headers are ASCII-only

Comment (sixaphone) [reply src/converter.js]: I might actually add custom support regarding headers, at leaste for standard ones so this will be reworked

Comment (ArtemDerevnjuk) [reply src/converter.js]: Let's skip this for now

File: src/converter.js [hunk 10]
```diff
@@ -224,7 +224,7 @@ var getHeadersArray = function(swagger, path, method) {
         const sample = OpenAPISampler.sample(param.schema || param, {}, swagger);
         headers.push({
           name: param.name,
-          value: encodeURIComponent((typeof sample === 'object') ? JSON.stringify(sample) : sample)
+          value: sample
```
Comment (ArtemDerevnjuk) [inline src/converter.js]: `(typeof sample === 'object') ? JSON.stringify(sample) : sample`

**Pull request merged**